Automattic/simplenote-macos
# New Metrics Screen: Mark 2

### Description:
In this PR we're implementing a mechanism that refreshes the Metrics UI, whenever the note onscreen is updated.

cc @danielebogo 
Sir!! you game for a fun PR?

Thank you!!

### Details:
- **EntityObserver:** New tool, useful to get callbacks whenever a NSManagedObject is updated
- **SimplenoteAppDelegate:** nullability specifiers

Ref. #458

### Test: Realtime Metrics
1. Log into your account
2. Click over any note
3. Click over the top right `( i )` button to reveal the Note Metrics
4. Edit the same Note in another device

- [x] Verify that the Note Metrics UI gets synchronized immediately
- [x] Verify that switching to Light / Dark mode refreshes the UI immediately (without closing the popover)

### Test: Multiple Selection
1. Log into your account
2. CMD + Click to select multiple notes
3. Click over the top right `( i )` button to reveal the Note Metrics

- [x] Verify that the Date fields display a `-` 
- [x] Verify that the Words / Characters fields display the sum of the values for each note

### Release
These changes do not require release notes.
